src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -301,11 +301,7 @@ private function parseXmlAttributes(\DOMNode $node)
         $data = array();
 
         foreach ($node->attributes as $attr) {
-            if (ctype_digit($attr->nodeValue)) {
-                $data['@'.$attr->nodeName] = (int) $attr->nodeValue;
-            } else {
-                $data['@'.$attr->nodeName] = $attr->nodeValue;
-            }
+            $data['@'.$attr->nodeName] = $attr->nodeValue;
```

Comment:
@jvasseur I changed in PR description BC break flag.
If checking possibility cast, then **only positive** integer will return as int and all others as string.
I think this behavior is wrong.